libs/vkd3d/device.c

```diff
@@ -4264,6 +4264,27 @@ static D3D12_TILED_RESOURCES_TIER d3d12_device_determine_tiled_resources_tier(st
     return D3D12_TILED_RESOURCES_TIER_2;
 }
 
+static D3D12_RESOURCE_HEAP_TIER d3d12_device_determine_heap_tier(struct d3d12_device *device)
+{
+    const VkPhysicalDeviceMemoryProperties *mem_properties = &device->memory_properties;
+    const struct vkd3d_memory_info *mem_info = &device->memory_info;
+    uint32_t i, host_visible_types = 0;
+
+    for (i = 0; i < mem_properties->memoryTypeCount; i++)
+    {
+        if (mem_properties->memoryTypes[i].propertyFlags & VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT)
+            host_visible_types |= 1u << i;
+    }
+
+    // Heap Tier 2 requires us to be able to create a heap that supports all resource
```

Comment:
Should perhaps look at bufferImageGranularity as well. D3D12 would require 64K alignment, but Vulkan can technically return 128K here.

Reply:
Done.